bundles/specmate-connectors/src/com/specmate/connectors/config/ProjectConfigService.java

```diff
@@ -228,6 +223,10 @@ public Object doChange() throws SpecmateException {
 			String projectLibraryKey = PROJECT_PREFIX + projectName + KEY_PROJECT_LIBRARY;
 			String[] libraryFolders = configService.getConfigurationPropertyArray(projectLibraryKey);
 			if (libraryFolders != null) {
+				Hashtable<String, Object> projectLibraryConfig = new Hashtable<String, Object>();

```

Comment:
This is probably incorrect. I guess what should be done here is to retrieve an existing configuration and update it. However, as the project services are created as configuartion factory, a new configuration is created with the library folders as only information. This will spawn a new project service without project name, whereas the original project service will not be updates with the library folders.

Reply:
no that is not it. The problem is that PROJECT_PID is not the PID of the service isntances but of the factory. The pids of the instances are automatically created by osgi and look like [PROJECT_PID]-[SOME-NUMBER]. Probably the right way would be to pick the service instance with a filter using the factory pid (PROJECT_PID) and the project name. 
Have a look here: https://stackoverflow.com/questions/30751407/how-can-single-instance-of-an-osgi-factory-configuration-be-read-from-java-in-cq

Reply:
Ok, I see. Can you check 3192ae4 if that solves the bug?